test/join.jl

```diff
@@ -1,8 +1,22 @@
 module TestJoin
     using Compat, Compat.Test, DataFrames
     using DataFrames: similar_missing
+    using TimeZones
+    import Base.==
+    import Base.hash
+    import Base.isequal
```

Comment:
Strictly speaking you don't need these `import`s, since you're qualifying the Base functions when you overload them.